### PR TITLE
scripts: remove board name from the identifier in list_boards.py

### LINKED ISSUE #69329: hwmv2: west boards output with identifiers contains board name as part of the identifier
~~It seems that `west boards --format "{identifiers}"` gives a python array output:~~
```
['stm32wb5mmg/stm32wb55xx']
['em_starterkit/emsk_em7d', 'em_starterkit/emsk_em9d', 'em_starterkit/emsk_em11d']
['emsdp/emsdp_em4', 'emsdp/emsdp_em5d', 'emsdp/emsdp_em6', 'emsdp/emsdp_em7d', 'emsdp/emsdp_em7d_esp', 'emsdp/emsdp_em9d', 'emsdp/emsdp_em11d']
['hsdk/arc_hsdk', 'hsdk/arc_hsdk/2cores']
['hsdk4xd/arc_hsdk4xd']
['iotdk/arc_iot']
['nsim/nsim_em', 'nsim/nsim_em7d_v22', 'nsim/nsim_em11d', 'nsim/nsim_hs', 'nsim/nsim_hs/sram', 'nsim/nsim_hs/smp', 'nsim/nsim_hs/flash_xip', 'nsim/nsim_hs/mpuv6', 'nsim/nsim_hs/hostlink', 'nsim/nsim_hs5x', 'nsim/nsim_hs5x/smp', 'nsim/nsim_hs5x/smp/12cores', 'nsim/nsim_hs6x', 'nsim/nsim_hs6x/smp', 'nsim/nsim_hs6x/smp/12cores', 'nsim/nsim_sem', 'nsim/nsim_sem/mpu_stack_guard', 'nsim/nsim_vpx5']
['robokit1/same70q21b']
['tlsr9518adk80d/tlsr9518']
```

~~Which does not go to stdout, it does not include any hwmv1 boards either~~

It seems that using the command `west boards --format "{identifiers}" | less` makes it look like there is no output, but there is output, but there is a huge amount of empty lines at the top (hwmv1 boards?) which should be fixed

### PR DESCRIPTION
**Pull request by tejlmand**
Fixes: #69329

The board name was printed as part of printing board identifiers because those was being concatenated in the `board_v2_identifiers()` method.

Keep the identifiers speparated from the board name and let the caller concatenate the strings when required.

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]
west completion files for bash/fish/zsh need updating to account for this

**Comment by 57300** [comment]
I guess this should be updated in light of #69905 to print `qualifiers`, or revert to printing what are now called `targets`.

**Comment by tejlmand** [comment]
> I guess this should be updated in light of #69905 to print `qualifiers`, or revert to printing what are now called `targets`.

to avoid n-PRs updating terminology, then the plan is a single PR updating all occurrences, hence the reason this PR does not address the terminology issue in existing places (identifiers --> qualifiers).

Will be done in dedicated PR.

File: scripts/west_commands/boards.py [hunk 1]
```diff
@@ -49,8 +49,9 @@ def do_add_parser(self, parser_adder):
             The following arguments are available:
 
             - name: board name
-            - identifiers: board identifiers
-            - arch: board architecture
+            - identifiers: board identifiers (will be empty for legacy boards)
```
Comment (nashif) [inline scripts/west_commands/boards.py:52]: we do not call this identifiers anymore, no?

Another thing, `west boards -f "{identifiers}"`  gives me a list of qualifiers, if a board has multiple, they are joined by a comma, so if I want to get the list of targets using "-f", it becomes very difficult. How about adding `{targets}`

Comment (nashif) [reply scripts/west_commands/boards.py:52]: ok, read the comment above re qualifiers, but rest of the above still applies.

Comment (tejlmand) [reply scripts/west_commands/boards.py:52]: > so if I want to get the list of targets using "-f", it becomes very difficult. How about adding {targets}

yes, but in this initial PR i didn't want to trigger this discussion again related to legacy vs new model boards:
https://github.com/zephyrproject-rtos/zephyr/pull/69740#discussion_r1522772483

Therefore this initial PR which simply ensures that issues are fixed and completion works.
Then a second PR allowing for longer discussion to take place.

I don't like to make an unnecessary block on the fix in this PR.

**Comment by tejlmand** [comment]
The compliance error:
``` R0801:Similar lines in 2 files
File:scripts/west_commands/runners/openocd.py
Line:1
Column:0
 R0801:Similar lines in 2 files
```
happens because a lot of duplicated code already exists between https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/intel_cyclonev.py and https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/openocd.py

**Comment by tejlmand** [comment]
> The compliance error:

Add `# pylint: disable=duplicate-code`

**Comment by nordicjm** [comment]
Rebase needed, documentation build fails and needs fixing

File: scripts/west_commands/completion/west-completion.fish [hunk 2]
```diff
@@ -196,19 +196,12 @@ function __zephyr_west_complete_help
...
-    end
-
-    # HWMv2
-    set -l boards (west 2>/dev/null boards --format="{identifiers}")
-    for board in $boards
-        set -l b (string split "," $board)
-        for variant in $b
-            printf "%s\n" $variant[1]
+        set -l b (string split "|" $board)
+        set -l identifiers (string split "," $b[2])
```
Comment (nordicjm) [inline scripts/west_commands/completion/west-completion.fish:202]: fish autoocomplete does not work (just blinks and does nothing)

File: scripts/west_commands/completion/west-completion.zsh [hunk 3]
```diff
@@ -102,9 +102,12 @@ _get_west_projs() {
 }
 
 _get_west_boards() {
-  _west_boards="$(__west_x boards --format={identifiers})\n$(__west_x boards --format={name})"
-  _west_boards=${_west_boards//$'\n'/\ }
-  _west_boards=${_west_boards//,/\ }
+  _west_boards=( $(__west_x boards --format='{name}|{identifiers}') )
+  for i in {1..${#_west_boards[@]}}; do
+    local name="${_west_boards[$i]%%|*}"
```
Comment (nordicjm) [inline scripts/west_commands/completion/west-completion.zsh:107]: zsh autocomplete does not work: `_get_west_boards:5: _west_boards: assignment to invalid subscript range`

File: scripts/west_commands/completion/west-completion.bash [hunk 4]
```diff
@@ -392,8 +392,13 @@ __set_comp_west_projs()
 
 __set_comp_west_boards()
 {
-	boards="$(__west_x boards --format={identifiers} "$@")\n$(__west_x boards --format={name} "$@")"
-	__set_comp ${boards//,/\ }
+	boards=( $(__west_x boards --format='{name}|{identifiers}' "$@") )
+	for i in ${!boards[@]}; do
```
Comment (nordicjm) [inline scripts/west_commands/completion/west-completion.bash:396]: bash autocomplete does not work

**Review by nordicjm — CHANGES_REQUESTED** [review]
Testing with PR rebased on `main`

**Comment by nordicjm** [comment]
Thinking about it, board revisions should probably be present in the autocomplete names too

**Comment by tejlmand** [comment]
> Testing with PR rebased on `main`

Rebased and adjusted to use qualifiers instead of identifiers.
The errors you noticed was due to the `--format` string was not adjusted to match the changes in #70438

**Comment by tejlmand** [comment]
> Thinking about it, board revisions should probably be present in the autocomplete names too

That's a good idea, and I think there was a similar comment not long ago, but forgot where it was.

Although the idea is good, and could be implemented for know revisions (those provided in board.yml), then we must consider how we want to handle `custom` revisions, as those may not be present in the board.yml.

This is a new possibility with HWMv2, which we couldn't do in HWMv1 because revisions was handled purely within CMake.

**Review by nordicjm — APPROVED** [review]

**Comment by tejlmand** [comment]
> This is a new possibility with HWMv2, which we couldn't do in HWMv1 because revisions was handled purely within CMake.

Created https://github.com/zephyrproject-rtos/zephyr/issues/70597

**Review by carlescufi — APPROVED** [review]

**Review by 57300 — APPROVED** [review]
LGTM, tested bash autocompletion.

**Pull request merged**